e2e/test/scenarios/dashboard-filters/dashboard-filters-sql-number.cy.spec.js

```diff
@@ -89,7 +87,73 @@ describe("scenarios > dashboard > filters > SQL > text/category", () => {
   });
 });
 
+describe("scenarios > dashboard > filters > SQL > number", () => {
+  const questionDetails = {
+    name: "Question 1",
+    native: {
+      query:
+        "SELECT * from products where true [[ and price > {{price}}]] [[ and rating > {{rating}} ]] limit 5;",
+      "template-tags": {
+        price: {
+          type: "number",
+          name: "price",
+          id: "b22a5ce2-fe1d-44e3-8df4-f8951f7921bc",
+          "display-name": "Price",
+        },
+        rating: {
+          type: "number",
+          name: "rating",
+          id: "b22a5ce4-fe1d-44e3-8df4-f8951f7921bc",
+          "display-name": "Rating",
+        },
+      },
+    },
+  };
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.createNativeQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { card_id, dashboard_id } }) => {
+        visitQuestion(card_id);
+
+        visitDashboard(dashboard_id);
+      },
+    );
+
+    editDashboard();
+  });
+
+  it("should keep filter value on blur", () => {
+    setupNumberFilter("Price");
+    setupNumberFilter("Rating");
+
+    saveDashboard();
+
+    cy.findByPlaceholderText("Price").type("95").blur();
+    cy.findByPlaceholderText("Rating").type("3.8").blur();
+
+    cy.findAllByTestId("table-row")
+      .should("have.length", 2)
+      .and("contain", "Doohickey")
+      .and("contain", "Widget");
+  });
+});
+
 function clearFilter(index) {
   filterWidget().eq(index).find(".Icon-close").click();
   cy.wait("@dashcardQuery2");
 }
+
+function setupNumberFilter(name) {
+  setFilter("Number", "Equal to");
+  cy.findByDisplayValue("Equal to").clear().type(name);
+
+  clickSelect();
+  popover().contains(name).click();
+}
+
+function clickSelect() {
+  cy.findByTestId("dashcard").findByText("Select…").click();
```

Comment:
We have `getDashboardCard` which supports selecting different dashcard when there are more than 2 dashcards on a dashboard. And its usage feels more established with over 100 results vs less than 10 for `findByTestId("dashcard")`

https://github.com/metabase/metabase/blob/120dc56cdb3b958b2d7d105b43b0c0bf28752025/e2e/support/helpers/e2e-dashboard-helpers.js#L14-L16

Reply:
I wasn't sure as it's not a real "dashcard", but a mapping form and I believe we use `getDashboardCard` to find a dashcard in view mode, not in edit